clang/lib/Parse/ParseDecl.cpp

```diff
@@ -4997,7 +5087,11 @@ void Parser::ParseStructUnionBody(SourceLocation RecordLoc,
 
   ParsedAttributes attrs(AttrFactory);
   // If attributes exist after struct contents, parse them.
-  MaybeParseGNUAttributes(attrs);
+  MaybeParseGNUAttributes(attrs, &LateFieldAttrs);
+
+  assert(!getLangOpts().CPlusPlus);
```

Comment:
Should we remove the assert? Are we restricting late parsing to C only?

Reply:
@delcypher C++ already has late parsing support for C++ classes and structs and I believe it's handled separately in `ParseDeclCXX.cpp`. I added the assertion to ensure that the code path isn't taken in C++. We should run clang tests if that breaks anything.

Reply:
Is it possible to combine C++'s late parsing with your implementation? Or is there little overlap?